### Exercise 2

Suppose the annual rainfall in a region follows a normal distribution with a mean of $\mu=800$ mm and a standard deviation of $\sigma=100$ mm.

Let $x$ be the random variable representing the annual rainfall in a randomly selected year. 

---
**(a)** Estimate the probability that the annual rainfall in a randomly selected year will be at least $1,000$ mm.

**Solution:**

To estimate the probability, we compute the Z-Score for $1,000$ mm as follows:

$$
Z = \frac{x-\mu}{\sigma} = \frac{1,000-800}{100}=2. 
$$

You can look up this Z-score in the standard normal distribution table, or use a software to approximate the probability $\mathbb{P}(Z\geq 2)$: 

In [2]:
from scipy.stats import norm

# Compute P(Z >= 2)
p = 1 - norm.cdf(2)

print(f'P(Z >= 2): {p:.4f}')

P(Z >= 2): 0.0228


The chance to observe an annual rainfall in the region of at least $1,000$ is approximately $2.3\%$. 

---
**(b)**
Compute the bounds provided by Markov's and Chebyshev's inequality. 

**Markov's Inequality**

We have 

$$
\mathbb{P}(x\geq 1,000) \leq \frac{\mu}{1000} = \frac{800}{1,000} = 0.8.
$$

According to Markov’s inequality, the probability that the annual rainfall in a randomly selected year will exceed $1,000$ mm is at most $0.8$ or $80\%$.

**Chebyshev's Inequality**

To use Chebyshev’s inequality, we bound $\mathbb{P}(|x - \mu| \geq t)$ instead of $\mathbb{P}(x\geq 1,000)$ and follow the line of argument of exercise 2b of sheet 2. The event $A = \{x \geq t\}$ can be equivalently expressed by 

$$
\begin{align*}
A 
&= \{x \geq t\} \\
&= \{x - \mu \geq t - \mu\}\\
&\subseteq \{x - \mu \geq t - \mu\} \cup \{x - \mu \leq -(t-\mu)\} \\
&= \{|x - \mu| \geq t - \mu\}\\
&= B 
\end{align*}
$$ 

Thus, $A \subseteq B$ implies $\mathbb{P}(A) \leq \mathbb{P}(B)$. Applying Chebyshev's inequality to $\mathbb{P}(B)$ gives
$$
\begin{align*}
\mathbb{P}(x \geq 1000) 
&\leq \mathbb{P}(|x - 800| \geq 200)\\
&\leq \frac{\sigma^2}{200^2}
= \left(\frac{100}{200}\right)^2 \\
&= 0.25.
\end{align*}
$$

According to Chebyshev's inequality, the probability that the annual rainfall in a randomly selected year will exceed $1,000$ mm is at most $0.25$ or $25\%$.

---
**(c)** Compute Chernoff's bound. For $s>0$, use 

$$
\begin{align*}
\mathbb{E}[\exp(sx)] = \exp\left(s\mu + \frac{1}{2}s^2\sigma^2\right).
\end{align*}
$$

Chernoff's bound is

$$
\begin{align*}
\mathbb{P}(x \geq 1,000) 
&\leq \frac{\mathbb{E}(\exp(sx))}{\exp(1,000s)}\\
&= \frac{\exp(s\mu + 0.5s^2\sigma^2)}{\exp(1,000s)}\\
&= \exp(s\mu + 0.5s^2\sigma^2- 1,000s)
\end{align*}
$$

The bound holds for all $s>0$. We choose $s$ to minimize the bound

$$
\exp(s\mu + 0.5s^2\sigma^2 - 1,000s).
$$

This is equivalent to minimize the quadratic function

$$
\begin{align*}
g(s) 
&= s\mu + 0.5s^2\sigma^2- 1,000s \\
&= 800s + 0.5\cdot (100s)^2 - 1,000s \\
&= 5,000s^2 -200s
\end{align*}
$$

Since $g$ is convex, we solve $g'(s) = 0$ for $s$:

1. The derivative is $g'(s) = 10,000s - 200 = 0$.
    
2. Solving $g'(s)=0$ for $s$ gives $s=0.02$. 

Thus, the minimum is $g(s) = 5,000s^2 -200s = -2$ giving the bound

$$
\mathbb{P}(x \geq 1,000) \leq \exp(-2) < 0.14.
$$

According to Chernoff's bound, the probability that the annual rainfall in a randomly selected year will exceed $1,000$ mm is at most $0.14$ or $14\%$.

---
**(c)** Compute Chernoff's bound. For $s>0$, use 

$$
\begin{align*}
\mathbb{E}[\exp(sx)] = \exp\left(s\mu + \frac{1}{2}s^2\sigma^2\right).
\end{align*}
$$

**Solution:**

Chernoff's bound is

$$
\begin{align*}
\mathbb{P}(x \geq t) 
&\leq \frac{\mathbb{E}[\exp(sx)]}{\exp(st)}
= \frac{\exp(s\mu + \frac{1}{2}s^2\sigma^2)}{\exp(st)}
= \exp\left(s\mu + \frac{1}{2}s^2\sigma^2 - st\right)
\end{align*}
$$

The bound holds for all $s>0$. We choose $s$ to minimize the bound

$$
\exp\left(\frac{1}{2}s^2\sigma^2 + (\mu  - t)s\right).
$$

This is equivalent to minimize the quadratic function

$$
g(s)=\frac{1}{2}s^2\sigma^2 + (\mu  - t)s.
$$

Since $g$ is convex, we solve $g'(s) = 0$ for $s$. The derivative is 

$$
g'(s) = s\sigma^2 + \mu - t.
$$.
    
Solving $g'(s)=0$ for $s$ gives 

$$
s_* = \frac{t-\mu}{\sigma^2} = \frac{1,000 - 800}{100^2} = 0.02.
$$

Then $g(s_*)$ is

$$
\begin{align*}
g(s_*) 
&= \frac{1}{2}s_*^2\sigma^2 + (\mu  - t)s_*\\
&= \frac{1}{2} (0.02\cdot 100)^2 -200 \cdot 0.02\\
&= 2 - 4 \\
&= -2
\end{align*}
$$

Inserting $g(s_*) = -2$ into the above Chernoff's bound gives

$$
\mathbb{P}(x \geq t) \leq \exp(-2) < 0.14.
$$

According to Chernoff's bound, the probability that the annual rainfall in a randomly selected year will exceed $1,000$ mm is at most $0.14$ or $14\%$.

**Comparison**

+ Actual: 2.2 %
+ Chernoff: 14 % 
+ Chebyshev: 25 %
+ Markov: 80 %